### PR TITLE
Backporting to 1.18.2

### PR DESCRIPTION
**Pull request by gniftygnome**
PR of work by [ScrewTSW](https://github.com/ScrewTSW).

> I am able to craft the portal frames, place it in the world, it auto-aligns and I'm able to fill it with eyes of ender and create the portal.
> 
> TODO: remove unnecessary TagLoadedEvent class

### CODE REVIEW

File: src/main/java/net/gnomecraft/obtainableend/data/ObtainableEndBlockLootTableProvider.java [hunk 1]
```diff
@@ -14,30 +14,28 @@
 import java.util.List;
 
 public class ObtainableEndBlockLootTableProvider extends FabricBlockLootTableProvider {
-    protected ObtainableEndBlockLootTableProvider(FabricDataOutput dataOutput) {
-        super(dataOutput);
-    }
```
Comment (gniftygnome) [inline src/main/java/net/gnomecraft/obtainableend/data/ObtainableEndBlockLootTableProvider.java:19]: Unnecessary style changes.

File: src/main/java/net/gnomecraft/obtainableend/data/ObtainableEndBlockLootTableProvider.java [hunk 2]
```diff
@@ -14,30 +14,28 @@
...
-                                                .properties(StatePredicate.Builder.create()
-                                                        .exactMatch(EndPortalFrameBlock.EYE, true)
-                                                )
-                                        )
-                                )
-                                .build()
-                        )
-                )
-        );
-    }
```
Comment (gniftygnome) [inline src/main/java/net/gnomecraft/obtainableend/data/ObtainableEndBlockLootTableProvider.java:42]: Unnecessary code style changes.

File: src/main/java/net/gnomecraft/obtainableend/tags/ObtainableEndTagModifier.java [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
-        TagsLoadedEvent.register(RegistryKeys.BLOCK, ObtainableEndTagModifier::updateBlockTags);
+        TagsLoadedEvent.register(Registry.BLOCK.getKey(), ObtainableEndTagModifier::updateBlockTags);
     }
 
     // We have to help the compiler understand the generics.
     private static <T> void updateBlockTags(TagManagerLoader.RegistryTags<T> registryTags) {
         if (!ObtainableEnd.getConfig().frameIsWitherImmune) {
             registryTags.tags().put(BlockTags.WITHER_IMMUNE.id(),
-                    registryTags.tags().get(BlockTags.WITHER_IMMUNE.id()).stream()
-                            .filter(entry -> entry.value() != Blocks.END_PORTAL_FRAME)
```
Comment (gniftygnome) [inline src/main/java/net/gnomecraft/obtainableend/tags/ObtainableEndTagModifier.java:19]: This definitely breaks the wither immunity disablement feature.

**Comment by gniftygnome** [comment]
> > TODO: remove unnecessary TagLoadedEvent class

Why is it unnecessary?  Aside from the fact I think the patch breaks the feature that uses this code...

I also need to review whether the `pickaxe_mineable` tag does what it should in 1.18.2, and the datagen overall but particularly the removal of the mod ID override on the recipe provider and whether the recipe gets placed in the `minecraft` namespace.

**Comment by ScrewTSW** [comment]
Related to #5

**Pull request merged**